src/MapPane.js

```diff
@@ -0,0 +1,158 @@
+/* @flow */
+import React, { PropTypes, Component } from 'react'
+import { forEach, omit, uniqueId, indexOf } from 'lodash'
+
+import childrenType from './types/children'
+import mapType from './types/map'
+import paneType from './types/pane'
+
+const BLACKLIST = ['tile', 'shadow', 'overlay', 'map', 'marker', 'tooltip', 'popup']
+
+export default class MapPane extends Component {
+  static propTypes = {
+    name: PropTypes.string,
+    children: childrenType,
+    map: mapType,
+    zIndex: PropTypes.number,
+    className: PropTypes.string,
+    style: PropTypes.object,
+    pane: paneType,
+  };
+
+  static contextTypes = {
+    map: mapType,
+    pane: paneType,
+  };
+
+  static childContextTypes = {
+    pane: paneType,
+  };
+
+  constructor ({name}) {
+    super()
+
+    this.state = {
+      _isMounted: false,
+    }
+
+    this._name = name || `pane-${uniqueId()}`
+  }
+
+  getChildContext () {
+    return {
+      pane: this._name,
+    }
+  }
+
+  componentWillMount () {
+    const map = this.context.map || this.props.map
+
+    if (this._name && map && map.createPane) {
+      const existing = this.getPane()
+
+      if (!existing) {
+        map.createPane(this._name, this.getParentPane())
+      }
+    }
+  }
+
+  componentWillReceiveProps (props) {
```

Comment:
I see, yes I missed the fact Leaflet is creating a new DOM node rather than using the one provided. It's too bad it doesn't allow it as we could have completely leveraged React for this, but in that case I think your implementation is needed as such.